medusa/providers/torrent/json/hdbits.py

```diff
@@ -44,48 +44,64 @@ def __init__(self):
         }
 
         # Proper Strings
+        self.proper_strings = []
 
         # Miscellaneous Options
 
         # Torrent Stats
+        self.minseed = None
+        self.minleech = None
 
         # Cache
-        self.cache = HDBitsCache(self, min_time=15)  # only poll HDBits every 15 minutes max
+        self.cache = tv.Cache(self, min_time=10)  # only poll HDBits every 10 minutes max
 
     def search(self, search_strings, age=0, ep_obj=None):
         """
         Search a provider and parse the results.
-
         :param search_strings: A dict with mode (key) and the search value (value)
         :param age: Not used
         :param ep_obj: Not used
         :returns: A list of search results (structure)
         """
-        # FIXME
+
         results = []
 
-        logger.log('Search string: {0}'.format(search_strings), logger.DEBUG)
+        logger.log('Search strings: {0}'.format(search_strings), logger.DEBUG)
 
         self._check_auth()
 
-        response = self.get_url(self.urls['search'], post_data=search_strings, returns='response')
-        if not response or not response.content:
-            logger.log('No data returned from provider', logger.DEBUG)
-            return results
-
-        if not self._check_auth_from_data(response):
-            return results
+        post_data = {
+            'username': self.username,
+            'passkey': self.passkey,
+            'category': [2, 3, 5],    # (1 Movie, 2 TV, 3 Documentary, 4 Music, 5 Sport, 6 Audio Track, 7 XXX, 8 Misc/Demo)
+        }
 
-        try:
-            jdata = response.json()
-        except ValueError:  # also catches JSONDecodeError if simplejson is installed
-            logger.log('No data returned from provider', logger.DEBUG)
-            return results
+        for mode in search_strings:
+            logger.log('Search mode: {0}'.format(mode), logger.DEBUG)
+
+            for search_string in search_strings[mode]:
+                if mode != 'RSS':
+                    logger.log('Search string: {search}'.format
+                                   (search=search_string), logger.DEBUG)
+                    post_data['search'] = search_string
+                response = self.get_url(self.urls['search'], post_data=json.dumps(post_data), returns='response')
+                if not response or not response.content:
+                    logger.log('No data returned from provider', logger.DEBUG)
+                    return results
+
+                if not self._check_auth_from_data(response):
+                    return results
+                try:
+                    jdata = response.json()
+                except ValueError:  # also catches JSONDecodeError if simplejson is installed
+                    logger.log('No data returned from provider', logger.DEBUG)
+                    return results
+
+                results += self.parse(jdata, None)
 
         results += self.parse(jdata, None)
```

Comment:
@durmato

Do like this: https://github.com/pymedusa/Medusa/blob/develop/medusa/providers/torrent/json/rarbg.py#L152-L154.com

Return items aligned with 'mode' loop